### PR TITLE
[WIP] Generating of TVB compatible datasets

### PR DESCRIPTION
**Pull request by sipv**
Issue #37.

### CODE REVIEW

File: tvb/recon/flow/surfaces_to_structural_datasets.py [hunk 1]
```diff
@@ -20,6 +21,131 @@ class Hemisphere(enum.Enum):
...
+            vert_norms[k, :] = np.mean(angle_scaling * triangle_normals[tri_list, :], axis=0)
+            # Scale by angle subtended.
+            vert_norms[k, :] = vert_norms[k, :] / np.sqrt(np.sum(vert_norms[k, :] ** 2, axis=0))
+            # Normalise to unit vectors.
+        except (ValueError, FloatingPointError):
+            # If normals are bad, default to position vector
+            # A nicer solution would be to detect degenerate triangles and ignore their
+            # contribution to the vertex normal
+            vert_norms[k, :] = vertices[k] / np.sqrt(vertices[k].dot(vertices[k]))
+            bad_normal_count += 1
```
Comment (maedoc) [inline tvb/recon/flow/surfaces_to_structural_datasets.py]: We shouldn't, in principle, have to handle this case. Did you encounter it?

File: tvb/recon/flow/surfaces_to_structural_datasets.py [hunk 2]
```diff
@@ -20,6 +21,131 @@ class Hemisphere(enum.Enum):
...
+    triangle_areas = triangle_areas[:, np.newaxis]
+    return triangle_areas
+
+
+def compute_region_areas(regions, triangle_areas, vertex_triangles, region_mapping):
+    """Compute the areas of given regions"""
+
+    region_surface_area = np.zeros(regions.size, 1)
+    avt = np.array(vertex_triangles)
+    # NOTE: Slightly overestimates as it counts overlapping border triangles,
```
Comment (maedoc) [inline tvb/recon/flow/surfaces_to_structural_datasets.py]: I wonder how FS handles that?  I guess their triangles are so small, they can exclude borders and be fine.

File: tvb/recon/flow/surfaces_to_structural_datasets.py [hunk 3]
```diff
@@ -124,18 +264,34 @@ def _get_subcortical_surfaces(self):
...
+        max_reg = max(np.max(regions_subcort), np.max(regions_cort))
+        orientations = np.zeros(max_reg + 1, 3)
+        areas = np.zeros(max_reg + 1, 1)
+
+        orientations[regions_subcort, :] = orients_subcort
+        orientations[regions_cort, :] = orients_cort
+        areas[regions_subcort] = areas_subcort
+        areas[regions_cort] = areas_cort
+
+        dataset = StructuralDataset(orientations, areas)
```
Comment (maedoc) [inline tvb/recon/flow/surfaces_to_structural_datasets.py]: should we be doing something with this `dataset` object?

File: tvb/recon/flow/surfaces_to_structural_datasets.py (hunk 1 above)
Comment (sipv) [inline tvb/recon/flow/surfaces_to_structural_datasets.py]: No, but I haven't even run it yet. The computations are mostly Tim's code that I've left as it was without changes

File: tvb/recon/flow/surfaces_to_structural_datasets.py (hunk 2 above)
Comment (sipv) [inline tvb/recon/flow/surfaces_to_structural_datasets.py]: You can quite easily scale the triangle contribution to the region area by the proportion of the vertices belonging to the region. Is it worth it? I don't know.

File: tvb/recon/flow/surfaces_to_structural_datasets.py (hunk 3 above)
Comment (sipv) [inline tvb/recon/flow/surfaces_to_structural_datasets.py]: Yeah, it is WIP for a reason.

Comment (maedoc) [inline tvb/recon/flow/surfaces_to_structural_datasets.py]: 👍  I'm brain dead today and will refrain from further comments

**Comment by maedoc** [comment]
Looks like a good start; do you have more commits on the way, or would you like me to get this finished?

**Comment by sipv** [comment]
Same as in #33 - works on its own, but it is not integrated.

**Comment by liadomide** [comment]
Outdated. Functionaly had been integrated already with similar code

**Pull request closed without merging**